play-services-basement/src/main/java/org/microg/safeparcel/SafeParcelReflectionUtil.java

```diff
@@ -364,6 +367,14 @@ private static void readField(AutoSafeParcelable object, Parcel parcel, int head
                 descriptor.field.set(object, l);
                 break;
             }
+            case Short: {
+                short i = (short) SafeParcelReader.readInt(parcel, header);
```

Comment:
This works because short and int are identical in SafeParcel's byte representation, but please use `SafeParcelReader.readShort()` nonetheless.

Reply:
Done.

Reply:
No need to cast to `short` anymore, as `readShort()` already returns a `short` :wink:

Reply:
Sorry about that, didn't pay enough attention the first time around.